NewForce-Cohort-6/tabloidfullstack-stuffing
# SB/3-View-All-Posts

# Description

Functionality added for a user to view a list of posts.

# Testing Instructions

Add seed data via the following query.
```SQL
INSERT INTO Post (Title, Content, ImageLocation, CreateDateTime, PublishDateTime, IsApproved, CategoryId, UserProfileId)
VALUES ('futurist post addendum mark', 'Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit.', 'http://lorempixel.com/920/360/', '2022-11-17', '2023-08-01', 1, 1, 2);
```

From branch `SB/3-View-All-Posts`

1. Login with any user
2. Navigate to Posts via navbar
3. You should see a list of posts ordered by post date descending with Title and Author displayed.
4. And the list should only contain approved Posts (ex. Post with title "engineer ubiquitous users" is an unapproved post and should not be seen)
5. Therefore you should not see the entry with title "engineer ubiquitous users" as it's an unapproved post (isApproved = 0).
6. And the list should only contain Posts with a publication date that is in the past
7. Therefore you should not see the entry with title "futurist post addendum mark" as it's publication date is in the future.

NOTE: Categories will be added to list once backend API dependency is complete (Robert)

# Checklist:
